dvc/repo/experiments/executor/base.py

```diff
@@ -464,6 +466,23 @@ def checkpoint_callback(
             exp_rev = cls.commit(
                 scm, exp_hash, exp_name=name, force=force, checkpoint=True
             )
+
+            git_remote = os.environ.get("DVC_EXP_AUTO_PUSH", None)
+            if git_remote:
+                from dvc.repo.experiments.push import push
+
+                branch = dvc.experiments.get_branch_by_rev(
+                    exp_rev, allow_multiple=None
+                )
+                branch_name = ExpRefInfo.from_ref(branch).name
```

Comment:
```suggestion
                branch = scm.get_ref(EXEC_BRANCH, follow=False)
```
We don't need to do the reverse rev->branch lookup from here, since we set the appropriate refs ourselves during exp execution (this `branch` will be the complete `refs/exps/...` refname and can be passed directly into `experiments.push`)